### PR TITLE
RELATED: SD-1220 Add type for opening delete dashboard dialog on KD

### PR DESCRIPTION
**Pull request by lvkhanh**
- Add typing for opening delete dashboard dialog on KD

---

Supported PR commands:

| Command         | Description            |
| --------------- | ---------------------- |
| `ok to test`    | Re-run standard checks |
| `extended test` | BackstopJS tests       |

---

# PR Checklist

-   [x] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [x] `check` passes
-   [x] `check-extended` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by lvkhanh** [comment]
extended test

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1524/ : SUCCESS in 5s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/3101/ : SUCCESS in 14m 13s

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1525/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/493/ : SUCCESS in 18m 23s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1527/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/3103/ : SUCCESS in 10m 42s

File: libs/sdk-embedding/src/iframe/kd.ts [hunk 1]
```diff
@@ -1272,4 +1282,45 @@ export namespace EmbeddedKpiDashboard {
...
+     * Open delete dashboard dialog, user will be able to delete the current existed dashboard
+     *
+     * Contract:
+     *
+     * if KD is currently editing dashboard, this command will try to open the dialog to delete the current existed dashboard,
+     *      on success DeleteDashboardDialogOpened will be posted
+     * commandFailed will be posted when:
+     *      KD is currently viewing dashboard or
+     *      No dashboard showing or
+     *      Donnot have permission delete existing object or
```
Comment (no23reason) [inline libs/sdk-embedding/src/iframe/kd.ts:1296]: Typo: Donnot. I would suggest rewording this to something like:
The current user does not have the permission to delete existing objects

File: libs/sdk-embedding/src/iframe/kd.ts [hunk 2]
```diff
@@ -1272,4 +1282,45 @@ export namespace EmbeddedKpiDashboard {
         GdcKdEventType.SetFilterParentsFailed,
         ISetFilterParentsFailedDataBody
     >;
+
+    /**
+     * Open delete dashboard dialog, user will be able to delete the current existed dashboard
```
Comment (no23reason) [inline libs/sdk-embedding/src/iframe/kd.ts:1287]: Typo: current existed. Should probably be "currently existing" (this is repeated also on line 1291).

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/497/ : SUCCESS in 31s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/757/ : SUCCESS in 16s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/3107/ : SUCCESS in 13m 52s
